CUE4Parse/UE4/Assets/Exports/StaticMesh/UStaticMesh.cs

```diff
@@ -0,0 +1,134 @@
+        public FStaticMeshRenderData RenderData { get; private set; }
+        public FStaticMaterial[]? StaticMaterials { get; private set; }
+        public Lazy<UMaterialInterface?>[]? Materials { get; private set; }
+
+        public override void Deserialize(FAssetArchive Ar, long validPos)
+        {
+            base.Deserialize(Ar, validPos);
+
+            FStripDataFlags stripDataFlags = Ar.Read<FStripDataFlags>();
+
+            bool bCooked = Ar.ReadBoolean();
+
+            var bodySetup = Ar.ReadObject<UObject>();
+
+            var navCollision = Ar.Ver >= UE4Version.VER_UE4_STATIC_MESH_STORE_NAV_COLLISION ? Ar.ReadObject<UObject>() : null;
+
+            if (!stripDataFlags.IsEditorDataStripped())
+            {
+                throw new NotImplementedException("Static Mesh with Editor Data not implemented yet");
+                // if (Ar.Ver < UE4Version.VER_UE4_DEPRECATED_STATIC_MESH_THUMBNAIL_PROPERTIES_REMOVED)
+                // {
+                //     var dummyThumbnailAngle = Ar.Read<FRotator>();
+                //     var dummyThumbnailDistance = Ar.Read<float>();
+                // }
+                // var highResSourceMeshName = Ar.ReadString();
+                // var highResSourceMeshCRC = Ar.Read<uint>();
+            }
+
+            LightingGuid = Ar.Read<FGuid>(); // LocalLightingGuid
+            Sockets = Ar.ReadArray(() => new FPackageIndex(Ar));
+
+            RenderData = new FStaticMeshRenderData(Ar, bCooked);
+
+            if (bCooked & Ar.Game >= EGame.GAME_UE4_20)
+            {
+                bool hasOccluderData = Ar.ReadBoolean();
+                if (hasOccluderData)
+                {
+                    Ar.ReadArray<FVector>(); // Vertices
+                    Ar.ReadArray<ushort>();  // Indics
+                }
+            }
+
+            if (Ar.Game >= EGame.GAME_UE4_14)
+            {
+                bool hasSpeedTreeWind = Ar.ReadBoolean();
+                if (hasSpeedTreeWind)
+                {
+                    Ar.Seek(validPos, SeekOrigin.Begin);
+                    return;
+                }
+                else
+                {
+                    if (FEditorObjectVersion.Get(Ar) >= FEditorObjectVersion.Type.RefactorMeshEditorMaterials)
+                    {
+                        // UE4.14+ - "Materials" are deprecated, added StaticMaterials
+                        StaticMaterials = Ar.ReadArray(() => new FStaticMaterial(Ar));
+                    }
+                }
+            }
+
+            if (StaticMaterials != null && StaticMaterials.Length != 0)
+            {
+                Materials = new Lazy<UMaterialInterface?>[StaticMaterials.Length];
+                for (int i = 0; i < StaticMaterials.Length; i++)
+                {
+                    Materials[i] = StaticMaterials[i].MaterialInterface;
+                }
+            }
+        }
+    }
+
+    public class UStaticMeshConverter : JsonConverter<UStaticMesh>
```

Comment:
the json writer format has changed, you can check out USkeleton.cs if you want an example of how it's done

Reply:
updated it